src/devel.rs

```diff
@@ -199,16 +199,35 @@ pub async fn gendb(config: &mut Config) -> Result<()> {
 }
 
 pub fn save_devel_info(config: &Config, devel_info: &DevelInfo) -> Result<()> {
-    create_dir_all(&config.cache_dir)
-        .with_context(|| format!("mkdir: {}", config.cache_dir.display()))?;
+    create_dir_all(&config.cache_dir).with_context(|| {
+        tr!(
+            "failed to create cache directory: {}",
+            config.cache_dir.display()
+        )
+    })?;
+
+    let mut temp = config.devel_path.to_owned();
+    temp.set_extension("json.tmp");
+
     let file = OpenOptions::new()
         .create(true)
         .write(true)
         .truncate(true)
-        .open(&config.devel_path);
-    let mut file = file.with_context(|| format!("open: {}", config.devel_path.display()))?;
+        .open(&temp);
+    let mut file =
+        file.with_context(|| tr!("failed to create temporary file: {}", temp.display()))?;
     let json = serde_json::to_string_pretty(&devel_info).unwrap();
-    file.write_all(json.as_bytes())?;
+
+    file.write_all(json.as_bytes())
+        .with_context(|| tr!("failed to write to temporary file: {}", temp.display()))?;
+
```

Comment:
I should have spotted it the first time but you probably need to close the file before rename so make sure it's flushed and the bytes really were wriitten.

Reply:
Good catch!